get-latest-version.py

```diff
@@ -35,58 +38,154 @@
-    if not github_issue_exists(issue_title):
-        issue = repo.create_issue(issue_title, "`%s` found in the import list, but not in the PPA. Not deployed yet or removed by accident?" % (component_name))
-        print("Package `%s` not found in elementary os-patches! - Created issue %d" % (component_name, issue.number))
-    sys.exit(0)
 
-patched_version = patched_sources[0].source_package_version
+def get_patched_sources():
+    """Get the current version of a package in elementary os patches PPA"""
+    return patches_archive.getPublishedSources(
+        exact_match=True,
+        source_name=component_name,
+        status="Published",
+        distro_series=series,
+    )
 
-# Search for a new version in the Ubuntu repositories
-pockets = ["Release", "Security", "Updates"]
-for pocket in pockets:
-    found_sources = ubuntu_archive.getPublishedSources(exact_match=True,
+
+def get_upstream_sources():
+    """Get the current version of a package in upstream PPA"""
+    return ubuntu_archive.getPublishedSources(
+        exact_match=True,
         source_name=component_name,
         status="Published",
         pocket=pocket,
-        distro_series=upstream_series)
-    if len(found_sources) > 0:
-        pocket_version = found_sources[0].source_package_version
+        distro_series=upstream_series,
+    )
+
+
+if len(get_patched_sources()) == 0:
+    issue_title = f"Package {component_name} not found in os-patches PPA"
+    if not github_pull_exists(issue_title):
+        issue = repo.create_issue(
+            issue_title,
+            f"{component_name} found in the import list, but not in the PPA. Not deployed yet or removed by accident?",
+        )
+        print(
+            f"Package {component_name} not found in elementary os-patches! - Created issue {issue.number}"
+        )
+    sys.exit(0)
+
+patched_version = get_patched_sources()[0].source_package_version
+
+# Search for a new version in the Ubuntu repositories
+for pocket in ["Release", "Security", "Updates"]:
+    upstream_sources = get_upstream_sources()
+    if len(upstream_sources) > 0:
+        pocket_version = upstream_sources[0].source_package_version
         if apt_pkg.version_compare(pocket_version, patched_version) > 0:
-            issue_title = "New version of %s available [%s]" % (component_name, upstream_series_name)
-            if not github_issue_exists(issue_title):
-                issue = repo.create_issue(issue_title, "The package `%s` in `%s` can be upgraded to version `%s`" % (component_name, upstream_series_name, pocket_version))
-                print("The patched package `%s` has a new version `%s` (was version `%s`) - Created issue %d" % (component_name, pocket_version, patched_version, issue.number))
+            pull_title = (
+                f"📦 Update {component_name} [{upstream_series_name}]"
+            )
+            if not github_pull_exists(pull_title):
+                base_branch = f"{component_name}-{upstream_series_name}"
```

Comment:
This works, but maybe we can avoid nested code here and write like this?

```py
    if len(upstream_sources) <= 0:
        sys.exit(0)

    pocket_version = upstream_sources[0].source_package_version
    if apt_pkg.version_compare(pocket_version, patched_version) <= 0:
        sys.exit(0)

    pull_title = (
        f"📦 Update {component_name} [{upstream_series_name}]"
    )
    if github_pull_exists(pull_title):
        sys.exit(0)

    base_branch = f"{component_name}-{upstream_series_name}"
```

Reply:
Thanks, that's valid suggestion.